$NetBSD: patch-CMakeLists.txt,v 1.3 2025/05/03 13:16:21 gdt Exp $

Avoid cmake erroring out with gcc7 at least.

Cherry-picked from https://github.com/OSGeo/PROJ/pull/4462/ which will
be in proj 9.6.1.

--- CMakeLists.txt.orig	2025-05-03 13:06:51.107368234 +0000
+++ CMakeLists.txt
@@ -355,7 +355,8 @@ endif()
 
 function (is_sharp_embed_available res)
     if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.21 AND
-        ((CMAKE_C_COMPILER_ID STREQUAL "GNU") OR (CMAKE_C_COMPILER_ID STREQUAL "Clang")))
+         ((CMAKE_C_COMPILER_ID STREQUAL "GNU" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 15.0) OR
+         (CMAKE_C_COMPILER_ID STREQUAL "Clang" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 19.0)))
         # CMAKE_C_STANDARD=23 only supported since CMake 3.21
         set(TEST_SHARP_EMBED
           "static const unsigned char embedded[] = {\n#embed __FILE__\n};\nint main() { (void)embedded; return 0;}"
